Fix block memory reads on slow targets.

timsifive opened the pull request:
The interesting new code concerns ignore_prev_addr and
this_is_last_read.

Additionally, I tweaked some debug output, and optimized
riscv_batch_run() when the batch is empty.

timsifive committed:
Fix block memory reads on slow targets.

The interesting new code concerns ignore_prev_addr and
this_is_last_read.

Additionally, I tweaked some debug output, and optimized
riscv_batch_run() when the batch is empty.

timsifive commented:
This doesn't fix everything either. :-(

timsifive commented:
But it does make things better, so please review so we can merge it.

p12nGH commented:
Do we have all the tests passing with this commit? What do you mean by it doesn't fix everything?

timsifive commented:
It passes all tests, when the machine running the tests is not heavily loaded. I'm still chasing out the details on when it fails. But in the meantime it's more reliable than it was.

p12nGH commented:
What hardware do you use?

p12nGH commented:
Nevermind, I realized we can only run tests on spike

timsifive commented:
You can run these tests against real hardware also, but I'm just working on spike right now. Real hardware doesn't become loaded in the same way.

timsifive commented:
Meanwhile, this change reduces the frequency of the problem, so I would still like it reviewed/merged.

mwachs5 commented:
Has this been tested on the HW Targets?

p12nGH commented:
@mwachs5 , no, I am  currently testing my binary build which doesn't have this commit. So far I didn't see any issues with real hardware.

timsifive committed:
Reset address if target was busy during bust write

Improve Issue #98.

DebugCompareSections is still failing for me (with an instrumented
sometimes-slow spike), but MemTestBlock now passes reliably.

timsifive commented:
riscv-tools build is failing, and this change would certainly improve things even if it won't fix anything.

Is there any reason nobody has approved it yet?

mwachs5 commented:
I think riscv-tools has already merged a bunch of openocd and riscv-tests fixes. Is it still failing?

timsifive commented:
https://travis-ci.org/riscv/riscv-tools says it is.

On Thu, Aug 24, 2017 at 4:19 PM, Megan Wachs <notifications@github.com>
wrote:

> I think riscv-tools has already merged a bunch of openocd and riscv-tests
> fixes. Is it still failing?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/riscv/riscv-openocd/pull/95#issuecomment-324783307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APYH7fLYZd8bwLGY7RHGGIwTKL-Xcpfbks5sbgUZgaJpZM4O29Pv>
> .
>

timsifive commented:
What is this waiting for?

BTW, it appears that this fixes everything as long as spike still has the debug_module debug printf()s compiled in (which it does in the version that riscv-tools is pointing at, but not in the latest riscv-isa-sim).

palmer-dabbelt reviewed (approved):
@timsifive Can you bump this to riscv-tools to see if it fixes the problems?
